### PR TITLE
Use the newest cairo lang release

### PR DESCRIPTION
**Pull request by war-in**
Closes #

## Introduced changes

- 
- 

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -103,36 +103,6 @@ jobs:
...
-            target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-      - name: Build compiler
-        if: steps.cache-rust.outputs.cache-hit != 'true'
-        working-directory: ./cairo
-        run: |
-          cargo build
-      - name: Create manifest file
-        run: |
-          readlink -f cairo/Cargo.toml >> starknet_py/tests/e2e/manifest-path
```
Comment (drknzz) [inline .github/workflows/checks.yml:135]: What would be the reason?

File: pyproject.toml [hunk 2]
```diff
@@ -22,8 +22,8 @@ crypto-cpp-py = "^1.2.0"
 marshmallow = "^3.15.0"
 marshmallow-oneofschema = "^3.0.1"
 typing-extensions = "^4.3.0"
-cairo-lang = { url = "https://github.com/starkware-libs/cairo-lang/releases/download/v0.11.0-pre/cairo-lang-0.11.0a0.zip" }
-eth-utils = "^2.1.0"
+cairo-lang = "^0.11.0"
+eth-utils = "<2.0.0"
```
Comment (drknzz) [inline pyproject.toml:26]: Was this needed by cairo-lang or?

File: starknet_py/tests/e2e/fixtures/devnet.py [hunk 3]
```diff
@@ -16,23 +14,6 @@ def get_available_port() -> int:
...
-    and create manifest-path file from manifest-path.template.
-    If should contain a path to top-level Cargo.toml file in Cairo 1 compiler directory.
-    """
-    try:
-        manifest_file_path = Path(os.path.dirname(__file__)) / "../manifest-path"
-        manifest = manifest_file_path.read_text("utf-8").splitlines()[0]
-
-        return ["--cairo-compiler-manifest", manifest]
-    except (IndexError, FileNotFoundError):
-        return []
```
Comment (drknzz) [inline starknet_py/tests/e2e/fixtures/devnet.py:33]: Does devnet resolve the compiler automatically or?

File: starknet_py/tests/e2e/mock/contracts_compiled/precompiled/minimal_contract_compiled.casm
Comment (drknzz) [inline starknet_py/tests/e2e/mock/contracts_compiled/precompiled/minimal_contract_compiled.casm:1]: I suppose these are the newest versions of the compiled contracts from cairo repo?

File: .github/workflows/checks.yml (hunk 1 above)
Comment (war-in) [reply .github/workflows/checks.yml:135]: image

File: pyproject.toml (hunk 2 above)
Comment (war-in) [reply pyproject.toml:26]: by devnet probably

**Review by drknzz — APPROVED** [review]

**Pull request merged**